include/utils/kahan.hpp

```diff
@@ -43,4 +45,94 @@ T kahan(InputIt first, InputIt last, T init) {
     return sum;
 }
 
+template<class InputIt, class T>
+T kahanBabushka(InputIt first, InputIt last, T init) {
+    T sum = init;
+    T cs = 0.;
+    T ccs = 0.;
+    T c = 0.;
+    T cc = 0.;
+
+    for (auto input = first; input != last; input++) {
+        T t = sum + *input;
+        if(std::abs(sum) >= std::abs(*input))
+            c = (sum - t) + *input;
+        else
+            c = (*input - t) + sum;
+        sum = t;
+        t = cs + c;
+        if(std::abs(cs) >= std::abs(c))
+            cc = (cs - t) + c;
+        else
+            cc = (c - t) + cs;
+        cs = t;
+        ccs = ccs + cc;
+    }
+    return sum + cs + ccs;
+}
+
+std::tuple<double,double,int,int> kahanBabushkaTuple(std::vector<BankedParticle>::iterator first, std::vector<BankedParticle>::iterator last) {
+
+    double sum1 = 0.;
+    double sum2 = 0.;
+
+    int sum3 = 0;
```

Comment:
I don't like the name of these variables, `sum1`, `sum2`, etc. They aren't descriptive. I would make the following changes: `sum1` => `sum_pos` and `sum2` => `sum_neg`. I would also change `sum3` to `count_pos` and `sum4` to `count_neg`. For the other variables, replace the 1 and 2 subscripts with `_pos` and `_neg`. This way we know exactly which goes to what.